src/db.php

```diff
@@ -1333,7 +1322,7 @@ private function prepare_directory()
          * @return mixed according to the query type
          * @see PDO::query()
          */
-        public function query($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, $arg3 = null, array $ctorargs = [])
+        public function query($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, mixed ...$ctorargs)
```

Comment:
The `mixed` type declaration requires PHP 8. This is a syntax error in all 5.x and 7.x versions of PHP.

Support for version 5.4 did not need to be removed. You could use [`func_get_args()`](https://www.php.net/manual/en/function.func-get-args.php) as an alternative to the splat (`...`) operator.

Reply:
@QWp6t I can't seem to find a signature that works with PHP 8 and PHP 5 that doesn't use the splat syntax. This is possible with PHP 5.6 but not 5.4. Of course we could use `func_get_args` for the actual values but that doesn't help us with the definition related errors.